### ISSUE TITLE
tutorials: confusing code sample and note about .dvc/cache dir

### ISSUE THREAD

**Issue by anferico** [iterative/dvc.org#806]
I was following the tutorial at [https://dvc.org/doc/tutorials/deep/preparation](https://dvc.org/doc/tutorials/deep/preparation) to start exploring DVC's functionalities. However, when I issue the command `dvc init` inside the `classify/` directory, I don't get the expected results.
In fact, `ls -a .dvc` should output the following:
`./      ../     .gitignore  cache/  config`
whereas my output is:
`./      ../     config .gitignore  tmp/`
so basically the `cache/` directory doesn't get created, but a `tmp/` directory does. This wouldn't be a big deal if the tutorial didn't mention that _"The .dvc/cache directory is one of the most important parts of any DVC repositories"_

**Comment by shcheklein** [comment]
@anferico yep, good point .. I think that we switched to creating it on-demand when we actually do some operations with it. @efiop could you confirm?

But the point stays - it's an important part. When you start doing `dvc add`, `dvc run`, etc ... you should see the directory.

**Comment by jorgeorpinel** [comment]
We should def. update the code sample and note though, so it's not confusing. This may affect other docs too. (I think that note is repeated in several places.)

**Comment by efiop** [comment]
@shcheklein I confirm :)

**Comment by jorgeorpinel** [comment]
@anferico do you think this version is less confusing? https://dvc-org-pr-810.herokuapp.com/doc/tutorials/deep/preparation

Note that the paragraph starting with "After DVC initialization..." I haven't updated yet because it's in 4 different documents, but I will.

**Comment by jorgeorpinel** [comment]
@anferico please check my PR #810 if you can. That should solve this confusion. Thanks again for reporting!
